javalib/src/main/scala/java/time/MonthDay.scala

```diff
@@ -0,0 +1,159 @@
+package java.time
+
+import java.time.chrono.IsoChronology
+
+import scala.scalajs.js
+
+import java.time.temporal._
+
+/**
+  * Created by alonsodomin on 22/12/2015.
+  */
+final class MonthDay private (month: Int, day: Int)
+    extends TemporalAccessor with TemporalAdjuster with Comparable[MonthDay]
+    with java.io.Serializable {
+
+  import Preconditions._
+  import ChronoField._
+
+  requireDateTime(month >= 1 && month <= 12, s"Invalid month value: $month")
+  requireDateTime(day >= 1 && day <= 31, s"Invalid day value: $day")
+
+  def isSupported(field: TemporalField): Boolean = field match {
+    case _: ChronoField => field == DAY_OF_MONTH || field == MONTH_OF_YEAR
+    case null           => false
+    case _              => field.isSupportedBy(this)
+  }
+
+  override def range(field: TemporalField): ValueRange = field match {
+    case DAY_OF_MONTH => ValueRange.of(1, getMonth().minLength(), getMonth().maxLength())
+    case _            => super.range(field)
+  }
+
+  // Implemented by TemporalAccessor
+  // def get(field: TemporalField): Int
+
+  def getLong(field: TemporalField): Long = field match {
+    case DAY_OF_MONTH  => day
+    case MONTH_OF_YEAR => month
+
+    case _: ChronoField =>
+      throw new UnsupportedTemporalTypeException(s"Field not supported: $field")
+
+    case _ => field.getFrom(this)
+  }
+
+  def getMonthValue(): Int = month
+
+  def getMonth(): Month = Month.of(month)
+
+  def getDayOfMonth(): Int = day
+
+  override def equals(other: Any): Boolean = other match {
+    case that: MonthDay =>
+      that.getMonthValue == getMonthValue() && that.getDayOfMonth == getDayOfMonth()
+    case _ => false
+  }
+
+  override def hashCode(): Int = {
+    val state = Seq(month, day)
+    state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
+  }
+
+  def isValidYear(year: Int): Boolean = {
+    if (month == 2 && day == 29)
+      IsoChronology.INSTANCE.isLeapYear(year)
+    else
+      true
+  }
+
+  def `with`(month: Month): MonthDay = {
+    require(month != null, s"'month' can not be null")
+    if (month == getMonth()) this
```

Comment:
Do you mean the `if` in the `isValidYear` method? Neither `if` or `else` clauses have braces already, I can add the braces to both if that is what you mean...